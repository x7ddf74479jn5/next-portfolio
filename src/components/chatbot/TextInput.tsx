import type { TextFieldProps } from "@material-ui/core/TextField";
import TextField from "@material-ui/core/TextField";
import React from "react";

type Props = TextFieldProps;

const TextInput: React.VFC<Props> = (props) => {
  return (
    <TextField
      fullWidth={true}
      id={props.id}
      label={props.label}
      margin={"dense"}
      multiline={props.multiline}
      rows={props.rows}
      value={props.value}
      type={props.type}
      onChange={props.onChange}
    />
  );
};

export default TextInput;
