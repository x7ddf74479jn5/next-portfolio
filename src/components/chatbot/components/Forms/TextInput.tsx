import type { TextFieldProps } from "@material-ui/core/TextField";
import TextField from "@material-ui/core/TextField";
import React from "react";

type Props = TextFieldProps;

const TextInput: React.VFC<Props> = (props) => {
  return (
    <TextField
      fullWidth={true}
      id="standard-basic"
      label={props.label}
      margin={"dense"}
      multiline={props.multiline}
      rows={props.rows}
      value={props.value}
      type={props.type}
      onChange={props.onChange}
    />
  );
};

export default TextInput;
