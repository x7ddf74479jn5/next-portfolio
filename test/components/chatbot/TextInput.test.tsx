import userEvent from "@testing-library/user-event";
import TextInput from "src/components/chatbot/TextInput";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("TextInput", () => {
  let renderResult: RenderResult;
  let inputElement: HTMLElement;
  let labelElement: HTMLElement;
  const mockHandleOnChange = jest.fn().mockReturnValue("typed");

  beforeEach(() => {
    renderResult = render(
      <TextInput id="id" label="label" multiline={false} rows={1} type="text" onChange={mockHandleOnChange} />
    );
    inputElement = renderResult.getByLabelText("label");
    labelElement = renderResult.getByText("label");
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(labelElement).toHaveTextContent("label");
    expect(inputElement).toHaveAttribute("id", "id");
    expect(inputElement).toHaveAttribute("rows", "1");
    expect(inputElement).toHaveAttribute("type", "text");
  });

  it("onChange event", () => {
    userEvent.type(inputElement, "test");
    expect(inputElement).toHaveValue("test");
    expect(mockHandleOnChange).toHaveBeenCalledTimes(4);
    userEvent.clear(inputElement);
    expect(inputElement).not.toHaveValue("test");
  });

  it("focus the input element when clicked the label element", () => {
    userEvent.click(labelElement);
    expect(inputElement).toHaveFocus();
  });
});
