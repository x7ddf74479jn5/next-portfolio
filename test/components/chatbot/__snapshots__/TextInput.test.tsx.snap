// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
      data-shrink="false"
      for="id"
      id="id-label"
    >
      label
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        id="id"
        rows="1"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
